apps/visibleplanets/visibleplanets.star

```diff
@@ -0,0 +1,399 @@
+    elif magnitude < 5:
+        description = "visible w/ binoculars"
+    elif magnitude < 6:
+        description = "near limit of naked eye"
+    elif magnitude < 10:
+        description = "visible with small telescope"
+    elif magnitude < 30:
+        description = "near limit of Hubble telescope"
+    else:
+        description = "you might need the James Webb Space telescope"
+
+    return description
+
+def get_cardinal_position_from_degrees(bearing):
+    """ Returns the cardinal position for a given bearing
+
+    Args:
+        bearing: in degrees
+    Returns:
+        The Cardinal position (N, NW, NE, S, SW, SE, E, W)
+    """
+
+    if bearing < 0:
+        bearing = 360 + bearing
+
+    # have bearning in degrees, not convert to cardinal point
+    compass_brackets = ["North", "NNE", "NE", "ENE", "East", "ESE", "SE", "SSE", "South", "SSW", "SW", "WSW", "West", "WNW", "NW", "NNW", "North"]
+    display_cardinal_point = compass_brackets[int(math.round(bearing // 22.5))]
+    return display_cardinal_point
+
+def get_body_position(body, location, check_time):
+    """ Gets the JSon Data from astronomyapi
+
+    Args:
+        body: Which celestial body we want info on
+        location: Location of the observer
+        check_time: Time to get the data for the given planet
+    Returns:
+        JSon Data
+    """
+
+    date_code = "%s-%s-%s" % (two_character_time_date_part(check_time.year), two_character_time_date_part(check_time.month), two_character_time_date_part(check_time.day))
+    tomorrow_date_code = check_time + time.parse_duration("24h")
+    tomorrow_date_code = "%s-%s-%s" % (two_character_time_date_part(tomorrow_date_code.year), two_character_time_date_part(tomorrow_date_code.month), two_character_time_date_part(tomorrow_date_code.day))
+    time_code = "%s:%s:%s" % (two_character_time_date_part(check_time.hour), two_character_time_date_part(check_time.minute), two_character_time_date_part(check_time.second))
+
+    params = {
+        "body": body,
+        "longitude": location["lng"],
+        "latitude": location["lat"],
+        "elevation": "0",
+        "from_date": date_code,
+        "to_date": tomorrow_date_code,
+        "time": time_code,
+    }
+
+    res = http.get(
+        url = "https://api.astronomyapi.com/api/v2/bodies/positions/%s" % body,
+        params = params,
+        headers = {
+            "Authorization": "Basic %s" % app_hash,
+        },
+    )
+
+    if res.status_code == 200:
+        return res.json()
+    else:
+        return None
+
+def two_character_time_date_part(number):
+    """ Returns two characters to represent an hour/minute/second needed when puttin together a timestamp 03:01:04
+
+    Args:
+        number: number we need to represent in two characters
+    Returns:
+        A two character string that represents the given number
+    """
+    if len(str(number)) == 1:
+        return "0" + str(number)
+    else:
+        return str(number)
+
+def get_bodies(app_hash):
```

Comment:
This function doesn't seem to called. Just in case it's supposed to be. =)

Reply:
removed -- I needed it to get a list of bodies I could request data on. Then hardcoded those bodies in the config.